node/src/documents.test.ts

```diff
@@ -33,11 +34,51 @@ test('create', async () => {
   )
 })
 
+/**
+ * Test of a workflow involving creating a temporary document
+ * and then specifying its format, and then later its path.
+ */
+ test('workflow-create-alter', async () => {
+  const doc = create();
+  expect(doc).toEqual(
+    expect.objectContaining({
+      temporary: true,
+      name: 'Unnamed',
+      format: expect.objectContaining({
+        name: "txt"
+      }),
+      status: 'synced'
+    })
+  )
+
+  expect(alter(doc.id, undefined, "md")).toEqual(
+    expect.objectContaining({
+      temporary: true,
+      name: 'Unnamed',
+      format: expect.objectContaining({
+        name: "md"
+      }),
+      status: 'synced'
+    }),
+  )
+
+  expect(alter(doc.id, "myfile.tex")).toEqual(
+    expect.objectContaining({
+      temporary: false,
```

Comment:
@nokome is this the intended behaviour, or is the documentation incorrect? Or third option, I'm confused 🙃 

>  * If a `path` is provided then the document's `name`, `format`, `status` and `temporary`
  * flag will be updated (`temporary` will be set to `true` and `status` to `unwritten`
  * immediately after the call).

Reply:
Docs are wrong, test is correct.